Fix ISO c++ does not allow assigning string literals to char*

peter-urban opened the pull request:
First partial fix for https://github.com/TEOS-10/GSW-C/issues/77

This fixes the c++ warnings related to assigning constant strings to char*
The fix is simple and works in c and c++ : 
- replaced several char* with const char* 

Most of the changes are unproblematic, since they only happen in gsw_check_functions.c

However, there is one change in gsw_saar_data.h (line 6,7 and 9). I wouldn't mind someone with better C command to double check this:
- const char** is a mutable pointer to a const char* and not a const pointer to mutable char* right?
- Also: While this is a minor change, it modifies the interface of gsw_get_version (const char** instead of char**). So it potentially require the libraries building on top of GSW-C to be updated? (e.g. GSW-Python)

efiring commented:
@peter-urban thank you for jumping on this.  But why are about 500,000 changes showing up in gsw_saar_data.h?

efiring commented:
gsw_get_version is not used in GSW-Python, and I doubt it is used anywhere else.
Nevertheless, we might as well get it right.  @mauzey1, I suspect you are the expert here,
so if you have time, please advise on this PR and #77.

peter-urban commented:
> @peter-urban thank you for jumping on this. But why are about 500,000 changes showing up in gsw_saar_data.h?

Oops, that was accidental auto-formatting. Should be fixed now.

efiring commented:
Googling turned up a nice utility program:

 $ cdecl explain "const char **x"
declare x as pointer to pointer to const char

peter-urban commented:
neat tool :-) In this case the function should work as intended.

I just modified gsw_check_functions.cpp line 146-160 to use 
the function once :1st_place_medal: It seems to work. The test header now reads:

```
============================================================================
 Gibbs SeaWater (GSW) Oceanographic Toolbox of TEOS-10 version 3.06.16 (C)
============================================================================

 These are the check values for the subset of functions that have been 
 converted into C from the Gibbs SeaWater (GSW) Oceanographic Toolbox 
 of TEOS-10 (version 3.06.16, released 15th_May_2011).
```

peter-urban commented:
Reverted the changes I mentioned in the last comment. 
This caused problems than I don't intend to fix in this PR.

FYI Problems: 
- to use the function I need to including gsw_saar_data.h in gsw_check_functions.cpp. 
- This causes multiple_declaration failures because gsw_get_version is neither declared static or inline.

peter-urban commented:
However, I believe the function does exactly the same as it did before this PR, so it should be fine

efiring commented:
@peter-urban what does your latest force-push do?  And do we need to keep all 3 commits, given that numbers 2 and 3 just cancel out? I could use the "squash and merge" option to simplify the history.

peter-urban commented:
Sorry for the force push. I just rebased the branch to the latest main not realizing that this may cause confusion after approval.
